src/value/use-on-change.ts

```diff
@@ -6,9 +6,9 @@ export function useOnChange<T>(
     value: MotionValue<T> | number | string,
     callback: Subscriber<T>
 ) {
-    useEffect(() =>
-        isMotionValue(value) ? value.onChange(callback) : undefined
-    )
+    useEffect(() => {
+        if (isMotionValue(value)) return value.onChange(callback)
+    }, [callback])
```

Comment:
Is this a driveby fix?

Reply:
Sort of, it didn't look right to me but I also haven't heard of any breakages that I'm fixing.